### PR TITLE
New builder

### PR DESCRIPTION
**Pull request by cchalmers**
Don't merge yet. See diagrams/diagrams-pandoc/pull/1.

### CODE REVIEW

File: diagrams-lib.cabal [hunk 1]
```diff
@@ -121,7 +122,9 @@ Library
                        unordered-containers >= 0.2 && < 0.2.6,
                        system-filepath >= 0.2 && < 0.5,
                        text >= 0.7.1 && < 1.3,
-                       mtl >= 2.0 && < 2.3
+                       mtl >= 2.0 && < 2.3,
+                       transformers,
+                       exceptions
```
Comment (byorgey) [inline diagrams-lib.cabal]: We ought to add some version bounds on these.

**Comment by cchalmers** [comment]
I added a 'findSandbox` function to `Diagrams.Util` which should be smart enough to find most sandbox (including remote sandboxes, sandboxes in parent folders or sandboxes from `DIAGRAMS_SANDBOX` variable). Other tools like builder and the linking finding in doc also use it. If anyone gets a chance to try it out let me know if it doesn't work for you.

I also made some changes to the CmdLine loop for a cleaner output.

I'm still unsure the about `BackendBuild` class.

**Comment by bergey** [comment]
@cchalmers Are you hoping to merge this before the 1.3 release, or immediately after?

I'm leaning towards after, but I'm willing to be persuaded.  As fiddly as the -doc build system is, I think I'd rather slog through one more release with a tested system, than change it with 3 weeks before we release.

**Comment by cchalmers** [comment]
I'm not sure. It shouldn't be too much work to get the site working using this. The htmls and pds generate the diagrams fine, it just needs integrating with the shake file and the website's style.

That said, I'm not happy with the details of `BackendBuild` and the whole thing feels unpolished.

Maybe it's best just to fix the current rst tutorials and work on this after.

I could split out the sandbox finder (0756912) and the prettier CmdLine loop (4857193) and merge that before 1.3.

**Comment by bergey** [comment]
Splitting out those two changes sounds great.

I'd like to think more about `Mainable` / `DiagramOpts` / `BackendBuild`.  Since you're also ambivalent about `BackendBuild`, doing it after the release seems better.

**Comment by bergey** [comment]
I think all of this has been merged except for the `BackendBuild` class.  Per https://github.com/diagrams/diagrams-builder/pull/16 we think we have a better way forward, so I'm closing this.  I'll leave the branch around in case I'm wrong about getting `diagrams-pandoc` to pick builders from `PATH`.

**Pull request closed without merging**